File: python_practice/pytest_practice/dice.py
from numpy import random

def roll(num_sides):
    try:
        num_sides_int = int(num_sides)
    except ValueError:
        return 'Please enter an integer.'
    if num_sides_int <= 0:
        return 'Please enter an integer greater than 0.'
    elif num_sides_int != float(num_sides):
        return 'Please enter an integer.'
    else:
        result = random.randint(1,num_sides_int+1)
        return result

if __name__ == '__main__':
    result = roll(input("Enter the number of sides: "))
    print(result)

File: cpp_practice/dice.cpp
// dice.cpp
#include <iostream>
#include <stdlib.h>
#include <time.h>


int roll(double sides) {
  if(std::cin.fail()) {
    std::cout << "Please enter an integer.\n";
  }
  else if(sides != int(sides)){
    std::cout << "Please enter an integer.\n";
  }
  else if(sides <= 0){
    std::cout << "Please enter an integer greater than 0.\n";
  }
  else {
    srand(time(NULL));
    std::cout << rand()%int(sides) + 1 << "\n";
  }
}

int main() {
  std::cout << "Please enter the number of sides: ";
  double sides;
  std::cin >> sides;
  roll(sides);
}
